src/pages/patch/PatchModal.tsx

```diff
@@ -0,0 +1,180 @@
+import { GET_PATCH_BUILD_VARIANTS } from "gql/queries/get-patch-build-variants";
+import { RESTART_PATCH } from "gql/mutations/restart-patch";
+import get from "lodash/get";
+import {
+  PatchBuildVariantsQuery,
+  PatchBuildVariantsQueryVariables,
+  RestartPatchMutation,
+  RestartPatchMutationVariables,
+} from "gql/generated/types";
+import { usePatchStatusSelect } from "hooks";
+import { useBannerDispatchContext } from "context/banners";
+import { TaskStatus } from "types/task";
+import { PatchBuildVariantAccordian } from "pages/patch/patchModal/index";
+
+const { gray } = uiColors;
+
+interface PatchModalProps {
+  visible: boolean;
+  onOk: () => void;
+  onCancel: () => void;
+}
+export const PatchModal: React.FC<PatchModalProps> = ({
+  visible,
+  onOk,
+  onCancel,
+}) => {
+  const {
+    successBanner,
+    errorBanner,
+    clearAllBanners,
+  } = useBannerDispatchContext();
+  const { id } = useParams<{ id: string }>();
+  const [shouldAbortInProgressTasks, setShouldAbortInProgressTasks] = useState(
+    false
+  );
+  const [restartPatch, { loading: mutationLoading }] = useMutation<
+    RestartPatchMutation,
+    RestartPatchMutationVariables
+  >(RESTART_PATCH, {
+    onCompleted: () => {
+      onOk();
+      successBanner(`Successfully restarted patch!`);
+    },
+    onError: (err) => {
+      onOk();
+      errorBanner(`Error while restarting patch: '${err.message}'`);
+    },
+  });
+  const { data, loading } = useQuery<
+    PatchBuildVariantsQuery,
+    PatchBuildVariantsQueryVariables
+  >(GET_PATCH_BUILD_VARIANTS, {
+    variables: { patchId: id },
+  });
+  const patchBuildVariants = get(data, "patchBuildVariants");
+  const [
+    selectedTasks,
+    validStatus,
+    { toggleSelectedTask, setValidStatus },
+  ] = usePatchStatusSelect(patchBuildVariants);
+
+  const handlePatchRestart = async (e): Promise<void> => {
+    e.preventDefault();
+    clearAllBanners();
+    try {
+      await restartPatch({
+        variables: {
+          patchId: id,
+          taskIds: selectedTasks,
+          abort: shouldAbortInProgressTasks,
+        },
+      });
+    } catch {
+      // This is handled in the onError handler for the mutation
+    }
+  };
+  return (
+    <Modal
+      title={<H2>Modify Version</H2>}
+      visible={visible}
+      onOk={onOk}
+      onCancel={onCancel}
+      footer={[
+        <Button onClick={onCancel}>Cancel</Button>,
+        <Button
+          data-cy="restart-patch-button"
+          disabled={selectedTasks.length === 0 || mutationLoading}
+          onClick={handlePatchRestart}
+          variant="danger"
+        >
+          Restart
+        </Button>,
+      ]}
+      width="50%"
+      wrapProps={{
+        "data-cy": "patch-modal",
+      }}
+    >
+      {!loading && patchBuildVariants && (
+        <>
+          <TreeSelect
+            onChange={setValidStatus}
+            state={validStatus}
+            tData={statusesTreeData}
+            inputLabel="Tasks Selected: "
```

Comment:
The tree select has a default label of "no filters selected" which doesn't make much sense in this context. Could you expose that default as a prop?